### ISSUE TITLE
App Settings should be persistent in db

### ISSUE THREAD

**Issue by khanhicetea** [her0e1c1/tango#14]
Hi @her0e1c1 ,

Each time I updated the app, my settings reset to default. Could them store in db ?

Thanks :)

**Comment by her0e1c1** [comment]
hi @khanhicetea thanks for your reporting and using my app.

I think I fixed this problem with version.
https://github.com/her0e1c1/tango/blob/master/src/index.tsx#L33

Can you tell me your app version on settings page? I think now it's 5.

**Comment by khanhicetea** [comment]
Yes, it's 5 now !

Thanks, I will check the next update and report to this issue :)

**Comment by her0e1c1** [comment]
Ah maybe when your app was updated to ver 5, it cleared your settings. I think no more it happened unless it becomes ver 6. Sorry for inconvenience ya ...

**Comment by khanhicetea** [comment]
Oh I see, so good !

**Comment by khanhicetea** [comment]
hi @her0e1c1 , may be the app still reset my settings at sometimes.

Btw, could we use first 5 chars on commit hash from `cat .git/refs/heads/master` as app version ? :smile:

**Comment by her0e1c1** [comment]
hi @khanhicetea thanks for reporting ... 
hmm for me a bit hard to understand this bug because my device is ok ...

I'm just worried about that hash commit can not use `>` and `<` operators.
Currently I use `==` only, so no problem. But I don't know the merit to convert ...

**Comment by khanhicetea** [comment]
ahh I see, so could we implement a new field as `commit hash` in setting page so we would know which commit the app is running :) If we have it, it's easier to testing and reporting bug !

**Comment by her0e1c1** [comment]
oh that's a good idea! OK :)

**Comment by her0e1c1** [comment]
Show git hash now ya. 39d402549e34ff1ec1d402337764894d40ca6202

**Comment by khanhicetea** [comment]
Ok I saw it !!! So good

I will track this issues next time the report to you soon

----------------
Khanh, Nguyen

**Comment by khanhicetea** [comment]
Hi @her0e1c1 ,

I know this is hard to debug my case, but it happened again

![photo6114090476949841925](https://user-images.githubusercontent.com/4528223/36936902-e033e40c-1f3d-11e8-81ac-ece7f6b98f15.jpg)

The gesture setting is reset to default

**Comment by her0e1c1** [comment]
sorry @khanhicetea today I upgraded version to 6
https://github.com/her0e1c1/tango/commit/736ae8bad3c6927b9a35cc4209e897cf21a7620b#diff-756e35fbe433c52dc16608c21c37138bR3

**Comment by khanhicetea** [comment]
So when you upgrade the version, it will reset settings, right ?

----------------
Khanh, Nguyen

**Comment by her0e1c1** [comment]
Yes, it resets, otherwise the app crashes. The version shows redux state schema.

I'm still thinking of how to fix ya
